lib/manager/terraform/lockfile/hash.ts

```diff
@@ -3,168 +3,129 @@ import extract from 'extract-zip';
-      const writeStream = fs.createWriteStream(downloadFileName);
-
-      let hash = null;
-      try {
-        await fs.pipeline(readStream, writeStream);
-
-        hash = await hashOfZipContent(downloadFileName, extractPath);
-        logger.trace(
-          { hash },
-          `Generated hash for ${build.name}-${build.version}`
-        );
-      } catch (err) {
-        /* istanbul ignore next */
-        logger.error({ err, build }, 'write stream error');
-      } finally {
-        // delete zip file
-        await fs.unlink(downloadFileName);
-      }
-      return hash;
-    },
-    { concurrency: 4 } // allow to look up 4 builds for this version in parallel
-  );
-  return hashes;
-}
+      // add double space, the filename and a new line char
+      rootHash.update('  ');
+      const fileName = file.replace(/^.*[\\/]/, '');
+      rootHash.update(fileName);
+      rootHash.update('\n');
+    }
 
-export async function createHashes(
-  repository: string,
-  version: string
-): Promise<string[]> {
-  // check cache for hashes
-  const repositoryRegexResult = repositoryRegex.exec(repository);
-  if (!repositoryRegexResult) {
-    // non hashicorp builds are not supported at the moment
-    return null;
-  }
-  const lookupName = repositoryRegexResult.groups.lookupName;
-  const backendLookUpName = `terraform-provider-${lookupName}`;
-
-  const cacheKey = `${TerraformProviderDatasource.defaultRegistryUrls[1]}/${repository}/${lookupName}-${version}`;
-  const cachedRelease = await packageCache.get<string[]>(
-    'terraform-provider-release',
-    cacheKey
-  );
-  // istanbul ignore if
-  if (cachedRelease) {
-    return cachedRelease;
+    rootHash.end();
+    const rootData = rootHash.read();
+    const result: string = rootData.toString('base64');
+    return result;
   }
-  let versionReleaseBackend: VersionDetailResponse;
-  try {
-    versionReleaseBackend = await getReleaseBackendIndex(
-      backendLookUpName,
-      version
-    );
-  } catch (err) {
-    logger.debug(
-      { err, backendLookUpName, version },
-      `Failed to retrieve builds for ${backendLookUpName} ${version}`
-    );
-    return null;
+
+  static async hashOfZipContent(
+    zipFilePath: string,
+    extractPath: string
+  ): Promise<string> {
+    await extract(zipFilePath, { dir: extractPath });
+    const files = await fs.readdir(extractPath);
+    // the h1 hashing algorithms requires that the files are sorted by filename
+    const sortedFiles = files.sort((a, b) => a.localeCompare(b));
+    const filesWithPath = sortedFiles.map((file) => `${extractPath}/${file}`);
+
+    const result = await TerraformProviderHash.hashFiles(filesWithPath);
+
+    // delete extracted files
+    await fs.rm(extractPath, { recursive: true });
+
+    return result;
   }
 
-  const builds = versionReleaseBackend.builds;
-  const hashes = await calculateHashes(builds);
+  static async calculateHashes(builds: TerraformBuild[]): Promise<string[]> {
```

Comment:
```suggestion

  @cache({
    namespace: `datasource-${TerraformProviderDatasource.id}-build-hashes`,
    key: (builds: TerraformBuild[]) => calcBuildsKey(builds), // TODO
    ttlMinutes: TerraformProviderHash.hashCacheTTL,
  })
  static async calculateHashes(builds: TerraformBuild[]): Promise<string[]> {
```

So cache the builds for a long time, or better cache every single build, so we only need to recompute missing build cache hits